blockchain/message_pool/src/msgpool/selection.rs

```diff
@@ -196,6 +208,97 @@ fn merge_and_trim(
     (result, gas_limit)
 }
 
+/// Like head_change, except it doesnt change the state of the MessagePool.
+/// It simulates a head change call.
+pub(crate) async fn run_head_change<T>(
+    api: &RwLock<T>,
+    pending: &RwLock<HashMap<Address, MsgSet>>,
+    from: Tipset,
+    to: Tipset,
+    rmsgs: &mut HashMap<Address, HashMap<u64, SignedMessage>>,
+) -> Result<(), Error>
+where
+    T: Provider + 'static,
+{
+    // TODO: This logic should probably be implemented in the ChainStore. It handles reorgs.
+    let mut left = Arc::new(from);
+    let mut right = Arc::new(to);
+    let mut left_chain = Vec::new();
+    let mut right_chain = Vec::new();
+    while left != right {
+        if left.epoch() > right.epoch() {
+            left_chain.push(left.as_ref().clone());
+            let par = api.read().await.load_tipset(left.parents()).await?;
+            left = par;
+        } else {
+            right_chain.push(right.as_ref().clone());
+            let par = api.read().await.load_tipset(right.parents()).await?;
+            right = par;
+        }
+    }
+    for ts in left_chain {
+        let mut msgs: Vec<SignedMessage> = Vec::new();
+        for block in ts.blocks() {
+            let (_, smsgs) = api.read().await.messages_for_block(&block)?;
+            msgs.extend(smsgs);
+        }
+        for msg in msgs {
+            add_to_selected_msgs(msg, rmsgs);
+        }
+    }
+
+    for ts in right_chain {
+        for b in ts.blocks() {
+            let (msgs, smsgs) = api.read().await.messages_for_block(b)?;
+
+            for msg in smsgs {
+                remove_from_selected_msgs(msg.from(), pending, msg.sequence(), rmsgs.borrow_mut())
+                    .await?;
+            }
+            for msg in msgs {
+                remove_from_selected_msgs(msg.from(), pending, msg.sequence(), rmsgs.borrow_mut())
+                    .await?;
+            }
+        }
+    }
+    Ok(())
+}
+
+/// This is a helper method for head_change. This method will remove a sequence for a from address
+/// from the messages selected by priority hashmap. It also removes the 'from' address and sequence from the MessagePool.
+pub(crate) async fn remove_from_selected_msgs(
```

Comment:
This method probably belongs with the msg pool module instead of here.

Reply:
@ec2 Thanks for the review. Could you explain why? They are not used anywhere in `msg_pool.rs`. The only places they are used as of now are in `selection.rs` and `mod.rs`? How about we keep it in `mod.rs`?

Reply:
You're right. mod.rs is better.